network/p2p/validators.go

```diff
@@ -71,20 +79,33 @@ func (v *Validators) refresh(ctx context.Context) {
 	v.lastUpdated = time.Now()
 }
 
+// Sample returns a random sample of connected validators
 func (v *Validators) Sample(ctx context.Context, limit int) []ids.NodeID {
 	v.lock.Lock()
 	defer v.lock.Unlock()
 
 	v.refresh(ctx)
 
-	return v.validatorIDs.Sample(limit)
+	sampled := make([]ids.NodeID, 0, limit)
+	validatorIDs := v.validatorIDs.Sample(limit)
```

Comment:
If you feel like this is an over optimization (not worth the decrease in readability), feel free to ignore.
```suggestion
	validatorIDs := v.validatorIDs.Sample(limit)
	sampled := validatorIDs[:0]
```